pkg/controller/sparkapplication/submission.go

```diff
@@ -133,6 +147,61 @@ func buildSubmissionCommandArgs(app *v1beta2.SparkApplication, driverPodName str
 	// Operator triggered spark-submit should never wait for App completion
 	args = append(args, "--conf", fmt.Sprintf("%s=false", config.SparkWaitAppCompletion))
 
+	if app.Spec.HadoopConfigMap != nil {
+		// Copy hadoop config files to operator pod
+		hadoopConfLoc, err := config.GetK8sConfigMap(app, *app.Spec.HadoopConfigMap)
```

Comment:
In the latest operator version, you could use hadoopConfigMap, it will export HADOOP_CONF_DIR  to driver or executor container automatic，so could del the code.